src/main/webapp/app/exercises/modeling/assess/modeling-assessment-editor/modeling-assessment-editor.component.ts

```diff
@@ -480,4 +486,21 @@ export class ModelingAssessmentEditorComponent implements OnInit {
     private removeHighlightedFeedbackOfColor(highlightedElements: Map<string, string>, color: string) {
         return new Map<string, string>([...highlightedElements].filter(([, value]) => value !== color));
     }
+    /**
+     * Calculates the total score of the current assessment.
+     * This function originally checked whether the total score is negative
+     * or greater than the max. score, but we decided to remove the restriction
+     * and instead set the score boundaries on the server.
+     */
+    calculateTotalScore() {
+        if (!this.feedback || this.feedback.length === 0) {
+            this.totalScore = 0;
+            return;
+        }
+        let totalScore = 0;
+        for (const feedback of this.feedback) {
+            totalScore += feedback.credits!;
+        }
+        this.totalScore = totalScore;
+    }
```

Comment:
The method `calculateTotalScore` could be shortened to:

```javascript
this.totalScore = (this.feedback || []).reduce((totalScore, feedback) => totalScore + feedback.credits!)
```

Reply:
I don't think this works, because inside reduce totalScore is of type feedback and not number.
So if we want to shorten it I will do something like that?
`const credits = (this.feedback || []).map((feedback) => feedback.credits);
        this.totalScore = credits.reduce((a, b) => a + b, 0);`

Reply:
I didn't actually test it. Looks like it works if the initial value is set to `0`:

![OnPaste 20200525-210403](https://user-images.githubusercontent.com/7109225/82838059-46b46a80-9ecb-11ea-84c9-a3ea374b02ad.png)